### PR TITLE
Remove consumer test related implementations

### PR DESCRIPTION
**Pull request by pstaeck**
SDCcc is not going to be used for consumer tests. Therefore, the purpose of the PR is to remove consumer test related implementations resp. move implementation of the test provider to test utils in order to use it for integration tests.

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Adherence to coding conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Adherence to javadoc conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* config update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* SDCcc executable ran against a test device (if necessary)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

File: sdccc/src/main/java/com/draeger/medical/sdccc/TestSuite.java [hunk 1]
```diff
@@ -368,8 +367,6 @@ private void performBasicMessagingCheck() {
     private Boolean setupDeviceAndProvider() {
         final var isConsumerEnabled =
                 injector.getInstance(Key.get(Boolean.class, Names.named(TestSuiteConfig.CONSUMER_ENABLE)));
-        final var isProviderEnabled =
-                injector.getInstance(Key.get(Boolean.class, Names.named(TestSuiteConfig.PROVIDER_ENABLE)));
 
         if (isConsumerEnabled) {
```
Comment (ldeichmann) [inline sdccc/src/main/java/com/draeger/medical/sdccc/TestSuite.java:371]: "Enable" for the consumer is now a redundant option as well, can be removed from the config file and here

File: sdccc/src/test/java/it/com/draeger/medical/sdccc/test_util/testprovider/TestProviderConfig.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * @see DefaultTestSuiteConfig
+ */
+public final class TestProviderConfig {
+
+    /*
+     * Provider configuration
+     */
+    private static final String PROVIDER = "Provider.";
+    public static final String PROVIDER_ENABLE = TestSuiteConfig.SDCCC + PROVIDER + Constants.ENABLE_SETTING_POSTFIX;
```
Comment (ldeichmann) [inline sdccc/src/test/java/it/com/draeger/medical/sdccc/test_util/testprovider/TestProviderConfig.java:25]: Is that really useful for the test provider?

**Comment by maximilianpilz** [comment]
@pstaeck the current repository config requires signed commits for merging onto main, you have unsigned commits in this PR

File: sdccc/src/main/java/com/draeger/medical/sdccc/TestSuite.java [hunk 3]
```diff
@@ -165,7 +165,7 @@ public long runTestSuite() {
         final SummaryGeneratingListener directSummary = new SummaryGeneratingListener();
         directTestLauncher.registerTestExecutionListeners(directSummary);
 
-        final Boolean isConsumerEnabled = setupDeviceAndProvider();
+        final Boolean isConsumerEnabled = setupDevice();
```
Comment (ldeichmann) [inline sdccc/src/main/java/com/draeger/medical/sdccc/TestSuite.java:168]: this boolean is also redundant, consumer is always enabled

File: sdccc/src/main/java/com/draeger/medical/sdccc/TestSuite.java [hunk 4]
```diff
@@ -165,13 +165,11 @@ public long runTestSuite() {
         final SummaryGeneratingListener directSummary = new SummaryGeneratingListener();
         directTestLauncher.registerTestExecutionListeners(directSummary);
 
-        final Boolean isConsumerEnabled = setupDevice();
```
Comment (ldeichmann) [inline sdccc/src/main/java/com/draeger/medical/sdccc/TestSuite.java]: Why did you move this into phase 1?? This is still the setup before running phase 1, i.e. generating messages.

**Review by ldeichmann — APPROVED** [review]
